### ISSUE TITLE
How to use cuseqlen correctly in `bertQKVToContextPlugin`?

### ISSUE THREAD

**Issue by linbinskn** [NVIDIA/TensorRT#2601]
Hi all, thanks for releasing efficient multihead attention kernels in TensorRT OSS. They are extremely fast!
However, I have met problem when I try to use the kernel in `bertQKVToContextPlugin` to support attention mask. I have tries several settings but all of them are failed.
1) cuseqlen.shape = [batch+1], qkv_ptr.shape = [batch, seq_len_max, head_num, head_size], where `cuseqlen` stores prefix of real seqlen and `qkv_ptr` is padding packed qkv(all seq_len padded into fix seq_len_max).
2) cuseqlen.shape = [batch+1], qkv_ptr.shape = [batch, seq_len_max, head_num, head_size], where `cuseqlen` stores real seqlen of each batch and `qkv_ptr` is padding packed qkv(all seq_len padded into fix seq_len_max).

I wonder know whether I use kernel by mistake to support attention_mask. What is the proper way to use them and what is the proper layout of qkv_ptr and cuseqlen.

**Comment by zerollzeng** [comment]
@ttyio ^ ^

**Comment by ttyio** [comment]
Hi @linbinskn ,

we use `sum(S)` for the `qkv_ptr`, and prefix sum for `cuseqlen` . let take example, we have 3 sequences:

     Sample #1: AAA
     Sample #2: BB
     Sampel #3: CCCC

for v2 plugin, the input `qkv_ptr` becomes:

    AAABBCCCC

the cuseqlen is:

    0 3 5 9

Hope this helps!

**Comment by linbinskn** [comment]
Very clear~ Thanks!
